docs/tutorials/parsec-tutorial.md

```diff
@@ -0,0 +1,557 @@
+
+Pseudo Applications:
+- **blackscholes:** Option pricing with Black-Scholes Partial Differential Equation (PDE)
+- **bodytrack:** Body tracking of a person
+- **facesim:** Simulates the motions of a human face
+- **ferret:** Content similarity search server
+- **fluidanimate:** Fluid dynamics for animation purposes with Smoothed Particle Hydrodynamics (SPH) method
+- **freqmine:** Frequent itemset mining
+- **raytrace:** Real-time raytracing
+- **swaptions:** Pricing of a portfolio of swaptions
+- **vips:** Image processing ([Project Website](https://github.com/libvips/libvips))
+- **x264:** H.264 video encoding ([Project Website](http://www.videolan.org/developers/x264.html))
+
+There are different sizes for possible inputs to each workload. Each size is explained below: 
+- **test:** very small set of inputs just to test the functionality of the program. 
+- **simdev:** small set of inputs intended to generate general behaviour of each program. Mainly used for simulators and development. 
+- **simsmall, simmedium, simlarge:** variable size inputs appropriate for testing microarchitectures with simulators. 
+- **native:** very large set of inputs intended for native execution. 
+ 
+This tutorial follows the following directory structure (inside the main directory):
+
+- configs-parsec-tests: gem5 run and configuration scripts to run PARSEC
+- disk-image: contains packer script and template files used to build a disk image.
+The built disk image will be stored in the same folder
+- gem5: gem5 [source code](https://gem5.googlesource.com/public/gem5) and the compiled binary
+- linux-stable: linux kernel [source code](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git)  used for full-system experiments
+- config.4.19.83: linux kernel config file used for its compilation
+- results: directory to store the results of the experiments (generated once gem5 jobs are executed)
+- launch_parsec_tests.py:  gem5 jobs launch script (creates all of the needed artifacts as well)
+
+
+## Setting up the environment
+First, we need to create the main directory named parsec-tests (from where we will run everything) and turn it into a git repository.
+Through the use of parsec-tests git repo, we will try to keep track of changes in those files which are not included in any git repo otherwise.
+An example of such files is gem5 run and config scripts (config-parsec-tests).
+We want to make sure that we can keep record of any changes in these scripts, so that a particular run of PARSEC benchmarks can be associated with a particular snapshot of these files.
+We also need to add a git remote to this repo pointing to a remote location where we want this repo to be hosted.
+
+```sh
+mkdir parsec-tests
+cd parsec-tests
+git init
+git remote add origin https://your-remote-add/parsec-tests.git
+```
+
+We also need to add a .gitignore file in our git repo, to avoid tracking those files which are not important or will be tracked through other git repos:
```

Comment:
@takekoputa, Did we create a template repo with this kind of stuff in it? Should we start writing these tutorials with that template repo in mind?

Reply:
@takekoputa and @mahyarsamani, can you sync up about the answer to this question?